commit fa21ddc158b5d7b5900856e5b131071302217a51
Author: NIIBE Yutaka <gniibe@fsij.org>
Date:   Mon Apr 10 11:45:00 2023 +0900

    random: Use getrandom only when it's appropriate.
    
    * random/rndgetentropy.c (_gcry_rndgetentropy_gather_random)
    [GRND_RANDOM]: Conditionalize the use of getrandom, as it's
    not a portable function.
    
    --
    
    Fixes-commit: aab1d63e4def41593312f76de016c885ffafecde
    GnuPG-bug-id: 6442
    Signed-off-by: NIIBE Yutaka <gniibe@fsij.org>

diff --git a/random/rndgetentropy.c b/random/rndgetentropy.c
index 513da0b9..a6f3c4ab 100644
--- a/random/rndgetentropy.c
+++ b/random/rndgetentropy.c
@@ -81,6 +81,7 @@ _gcry_rndgetentropy_gather_random (void (*add)(const void*, size_t,
       do
         {
           _gcry_pre_syscall ();
+#ifdef GRND_RANDOM
           if (fips_mode ())
             {
               /* DRBG chaining defined in SP 800-90A (rev 1) specify
@@ -98,6 +99,7 @@ _gcry_rndgetentropy_gather_random (void (*add)(const void*, size_t,
               ret = getrandom (buffer, nbytes, GRND_RANDOM);
             }
           else
+#endif
             {
               nbytes = length < sizeof (buffer) ? length : sizeof (buffer);
               ret = getentropy (buffer, nbytes);
